app/controllers/concerns/vet360/writeable.rb

```diff
@@ -40,11 +38,13 @@ def invalidate_cache
 
     def build_record(type, params)
       # This needs to be refactored after V2 upgrade is complete
-      model = if type == 'address' && Flipper.enabled?(:va_v3_contact_information_service, @current_user)
-                'VAProfile::Models::V3::Address'
-              else
-                "VAProfile::Models::#{type.capitalize}"
-              end
+      if type == 'address' && Flipper.enabled?(:va_v3_contact_information_service, @current_user)
+        model = 'VAProfile::Models::V3::Address'
+        # Ensures the address_pou is valid
+        params[:address_pou] = 'RESIDENCE' if params[:address_pou] == 'RESIDENCE/CHOICE'
```

Comment:
Found one error: Address_pou comes in as 'RESIDENCE/CHOICE' from the frontend. Contact Information V2 requires RESIDENCE not RESIDENCE/CHOICE.